gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/architecture/CoreRulesTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package architecture;
+
+import static com.tngtech.archunit.base.DescribedPredicate.not;
+import static com.tngtech.archunit.core.domain.JavaClass.Predicates.resideInAPackage;
+import static com.tngtech.archunit.lang.conditions.ArchConditions.haveNameMatching;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.classes;
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Yann TAVERNIER (yann.tavernier at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+public class CoreRulesTest extends AbstractApimArchitectureTest {
+
+    /**
+     * Core package should not depend on any framework
+     * - Spring
+     * - what's under io.gravitee.apim.infra
+     */
+    @Test
+    public void core_package_should_not_depend_on_framework() {
```

Comment:
Is it possible to build the "reverse" rule: 

> core package should only depend on itself. 

We may need to authorize some other packages (like lombok or slf4j) but I would add explicitly those exception 🤔

Reply:
refactored to be more explicit